aiida/orm/nodes/data/array/bands.py

```diff
@@ -825,7 +825,10 @@ def _prepare_mpl_singlefile(self, *args, **kwargs):
 
         s_header = matplotlib_header_template.substitute()
         s_import = matplotlib_import_data_inline_template.substitute(all_data_json=json.dumps(all_data, indent=2))
-        s_body = matplotlib_body_template.substitute()
+        if len(all_data['paths']) == 1:
```

Comment:
It seem to me that `_prepare_mpl_singlefile` could simply be reusing `_prepare_mpl_withjson` in order to avoid code duplication. Is this correct?

Reply:
Yes, these two can be combined. I'll try.

Reply:
I tried, however, each of `_prepare_mpl_*` function is similar but slightly different. I can't think of a better way to combine them at the moment. Using a lone template string and leave all optional string as arguments seems like not a good idea.